javav2/example_code/redshift/src/main/java/com/example/scenario/RedshiftScenario.java

```diff
@@ -0,0 +1,550 @@
+        JsonParser parser = new JsonFactory().createParser(new File(fileName));
+        com.fasterxml.jackson.databind.JsonNode rootNode = new ObjectMapper().readTree(parser);
+        Iterator<JsonNode> iter = rootNode.iterator();
+        ObjectNode currentNode;
+        int t = 0;
+        while (iter.hasNext()) {
+            if (t == number)
+                break;
+            currentNode = (ObjectNode) iter.next();
+            int year = currentNode.get("year").asInt();
+            String title = currentNode.get("title").asText();
+
+            // Use SqlParameter to avoid SQL injection.
+            List<SqlParameter> parameterList = new ArrayList<>();
+            String sqlStatement = "INSERT INTO Movies VALUES( :id , :title, :year);";
+
+            // Create the parameters.
+            SqlParameter idParam = SqlParameter.builder()
+                .name("id")
+                .value(String.valueOf(t))
+                .build();
+
+            SqlParameter titleParam= SqlParameter.builder()
+                .name("title")
+                .value(title)
+                .build();
+
+            SqlParameter yearParam = SqlParameter.builder()
+                .name("year")
+                .value(String.valueOf(year))
+                .build();
+            parameterList.add(idParam);
+            parameterList.add(titleParam);
+            parameterList.add(yearParam);
+
+            try {
+                ExecuteStatementRequest insertStatementRequest = ExecuteStatementRequest.builder()
+                    .clusterIdentifier(clusterId)
+                    .sql(sqlStatement)
+                    .database(databaseName)
+                    .dbUser(userName)
+                    .parameters(parameterList)
+                    .build();
+
+                redshiftDataClient.executeStatement(insertStatementRequest);
+                System.out.println("Inserted: " + title + " (" + year + ")");
+                t++;
+
+            } catch (RedshiftDataException e) {
+                System.err.println("Error inserting data: " + e.getMessage());
+                System.exit(1);
+            }
+        }
+        System.out.println(t + " records were added to the Movies table. ");
+    }
+    // snippet-end:[redshiftdata.java2.add.record.main]
+
+    // snippet-start:[redshiftdata.java2.checkstatement.main]
+    public static void checkStatement(RedshiftDataClient redshiftDataClient, String sqlId) {
+        try {
+            DescribeStatementRequest statementRequest = DescribeStatementRequest.builder()
+                .id(sqlId)
+                .build();
+
+            String status;
+            while (true) {
+                DescribeStatementResponse response = redshiftDataClient.describeStatement(statementRequest);
+                status = response.statusAsString();
+                System.out.println("..." + status);
```

Comment:
I entered 2010 and got a statement status FAILED, but there is no way out of this loop for a failure, so the program hangs indefinitely.

Reply:
Updated Logic to address this. Nice comment.